src/components/time-picker/time-picker.tsx

```diff
@@ -0,0 +1,382 @@
+        const {label, name, placeholder, disabled, format} = this.props;
+        const currentIsTime = isTimeSegment(currentSegment);
+        const currentSegmentValue = this.state[currentSegment] || 0;
+
+        return (
+            <div
+                data-automation-id="TIME_PICKER"
+                cssStates={{focus, disabled: disabled!}}
+            >
+                <input
+                    data-automation-id="TIME_PICKER_INPUT"
+                    type="text"
+                    tabIndex={isTouch ? -1 : 0}
+                    className="input"
+                    ref={elem => this.input = elem}
+                    value={inputValue}
+                    placeholder={placeholder || defaultValue(format!)}
+                    disabled={disabled}
+                    name={name}
+                    aria-label={label}
+                    onChange={this.onInputChange}
+                    onFocus={this.onInputFocus}
+                    onBlur={this.onInputBlur}
+                    onKeyDown={this.onInputKeyDown}
+                    onMouseDown={this.onInputMouseDown}
+                    onClick={this.onInputClick}
+                />
+                {!isTouch && !disabled && inputValue &&
+                    <Stepper
+                        className="stepper"
+                        disableUp={
+                            currentIsTime &&
+                            !isValidValue(currentSegmentValue + 1, currentSegment as any, ampm)
+                        }
+                        disableDown={
+                            currentIsTime &&
+                            !isValidValue(currentSegmentValue - 1, currentSegment as any, ampm)
+                        }
+                        onUp={this.createStepperHandler(+1)}
+                        onDown={this.createStepperHandler(-1)}
+                    />
+                }
+                {isTouch &&
+                    <label className="label">
+                        <input
+                            className="native-input"
+                            type="time"
+                            name={name}
+                            aria-label={label}
+                            value={this.getValue()}
+                            disabled={disabled}
+                            onFocus={this.onNantiveInputFocus}
+                            onBlur={this.onInputBlur}
+                            onChange={this.onNativeInputChange}
+                        />
+                    </label>
+                }
+            </div>
+        );
+    }
+
+    private getInitialState(props: Props) {
+        const segments = propsValueToSegments(props.value, props.format);
+        return {
+            ...segments,
+            inputValue: segmentsToInputValue(segments)
+        };
+    }
+
+    private getValue() {
+        const {hh, mm, ampm} = this.state;
+        return [
+            to24(hh || 0, ampm),
+            mm || 0
```

Comment:
I don't think it is a good idea, since `to24` is really simple convertor which expect correct input. You will newer want `Math.sin` to have default value to `0`